From 56cac0b3ca8cca5e48c5da4bd714b2b0b3918948 Mon Sep 17 00:00:00 2001
From: Pluto Yang <yangyj.ee@gmail.com>
Date: Mon, 29 Jul 2024 17:29:36 +0800
Subject: [PATCH 2/5] drm/stm: dsi: use clk_get_rate to set pll_in_khz

The parent of 'ck_dsi_phy' for stm32f4 devices is not properly
defined. The parent_rate passed by clock framework is always 0.
Use clk_get_rate to set pll_in_khz is a temporary solution.
---
 drivers/gpu/drm/stm/dw_mipi_dsi-stm.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/stm/dw_mipi_dsi-stm.c b/drivers/gpu/drm/stm/dw_mipi_dsi-stm.c
index a3eae5b2c26e..e33c83d68e0a 100644
--- a/drivers/gpu/drm/stm/dw_mipi_dsi-stm.c
+++ b/drivers/gpu/drm/stm/dw_mipi_dsi-stm.c
@@ -257,7 +257,7 @@ static unsigned long dw_mipi_dsi_clk_recalc_rate(struct clk_hw *hw,
 
 	DRM_DEBUG_DRIVER("\n");
 
-	pll_in_khz = (unsigned int)(parent_rate / 1000);
+	pll_in_khz = clk_get_rate(dsi->pllref_clk) / 1000;
 
 	val = dsi_read(dsi, DSI_WRPCR);
 
@@ -282,7 +282,7 @@ static int dw_mipi_dsi_clk_determine_rate(struct clk_hw *hw,
 
 	DRM_DEBUG_DRIVER("\n");
 
-	pll_in_khz = (unsigned int)(req->best_parent_rate / 1000);
+	pll_in_khz = clk_get_rate(dsi->pllref_clk) / 1000;
 
 	/* Compute best pll parameters */
 	idf = 0;
@@ -312,7 +312,7 @@ static int dw_mipi_dsi_clk_set_rate(struct clk_hw *hw, unsigned long rate,
 
 	DRM_DEBUG_DRIVER("\n");
 
-	pll_in_khz = (unsigned int)(parent_rate / 1000);
+	pll_in_khz = clk_get_rate(dsi->pllref_clk) / 1000;
 
 	/* Compute best pll parameters */
 	idf = 0;
@@ -425,7 +425,7 @@ dw_mipi_dsi_get_lane_mbps(void *priv_data, const struct drm_display_mode *mode,
 	unsigned int pll_in_khz, pll_out_khz;
 	int ret, bpp;
 
-	pll_in_khz = (unsigned int)(clk_get_rate(dsi->pllref_clk) / 1000);
+	pll_in_khz = clk_get_rate(dsi->pllref_clk) / 1000;
 
 	/* Compute requested pll out */
 	bpp = mipi_dsi_pixel_format_to_bpp(format);
-- 
2.55.0

